server/ai_http.go

```diff
@@ -271,6 +274,29 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 			return
 		}
 
+		// Precreate the channels to avoid race conditions
+		// TODO get the expected mime type from the request
+		pubCh := trickle.NewLocalPublisher(h.trickleSrv, mid, "video/MP2T")
+		pubCh.CreateChannel()
+		subCh := trickle.NewLocalPublisher(h.trickleSrv, mid+"-out", "video/MP2T")
+		subCh.CreateChannel()
+
+		// Subscribe to the publishUrl for payments monitoring
+		go func() {
+			sub := trickle.NewLocalSubscriber(h.trickleSrv, mid)
+			for {
+				segment, err := sub.Read()
+				if err != nil {
+					clog.Infof(ctx, "Error getting local trickle segment err=%v", err)
+					return
+				}
+				// We can do something with the segment data here
+				io.Copy(io.Discard, segment.Reader)
```

Comment:
We can check the payment here, but let's say the payment check fails (so Gateway hasn't paid for the stream). How do we stop all the processing from here?

Reply:
there's a few things (but I think we can still get basic payments in without waiting on these)

- channels needs to be closed on the orch side - we can use the local publisher which already has a `Close()` method or add an API directly on the server for that
- runner needs to be notified - either via end-of-stream which is sent down the subscribe leg when a channel is closed, or explicit job RPC
- we need to prevent the same channels from being recreated on the O when the next publish comes in, so need to add either a local registry + auth callback to the trickle server, or disallow implicit channel creation via POST and keep pre-creating everything on the O locally.
- and probably some end-of-stream indicator on the publish leg, to send back to the gateway